### PR TITLE
Raising for invalid dtype issue #15520

### LINKED ISSUE #15520: BUG: invalid dtypes should raise
xref https://github.com/dask/dask/issues/1965

when an explict dtype is provided, we should raise if its not a valid ``numpy/pandas`` dtype.

```
In [1]: pd.Series([], name='time',dtype=pd.Timestamp)
Out[1]: Series([], Name: time, dtype: object)
```

### PR DESCRIPTION
**Pull request by petioptrv**
closes #15520 

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes git diff upstream/master | flake8 --diff
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -172,6 +172,12 @@ def _validate_dtype(self, dtype):
                 raise NotImplementedError("compound dtypes are not implemented"
                                           "in the {0} constructor"
                                           .format(self.__class__.__name__))
+            
+            # check if coerced dtype is of type object
```
Comment (jreback) [inline pandas/core/generic.py]: use ``pandas.types.common.pandas_dtype``

**Review by jreback — CHANGES_REQUESTED** [review]
first thing to always do are tests

**Comment by petioptrv** [comment]
Are you saying I should raise within the pandas_dtype method?

**Comment by jreback** [comment]
yes ``pandas_dtype`` will raise if an invalid string is passed, but not an invalid object

```
In [1]: pandas.types.common.pandas_dtype(pd.Timestamp)
Out[1]: dtype('O')

In [2]: pandas.types.common.pandas_dtype('foo')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-07647ef910a7> in <module>()
----> 1 pandas.types.common.pandas_dtype('foo')

/Users/jreback/pandas/pandas/types/common.py in pandas_dtype(dtype)
    491         return dtype
    492 
--> 493     return np.dtype(dtype)

TypeError: data type "foo" not understood

```

**Comment by jreback** [comment]
if you want to add some tests ping and will reopen

**Pull request closed without merging**